browser/ai_chat/ai_chat_settings_helper.cc

```diff
@@ -103,10 +133,69 @@ void AIChatSettingsHelper::GetManageUrl(GetManageUrlCallback callback) {
 #if defined(OFFICIAL_BUILD)
   std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart);
 #else
-  std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart, brave_domains::STAGING);
+  std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart,
+                                                        brave_domains::STAGING);
 #endif
 
-  std::move(callback).Run(base::StrCat({url::kHttpsScheme, url::kStandardSchemeSeparator, domain}));
+  std::move(callback).Run(
+      base::StrCat({url::kHttpsScheme, url::kStandardSchemeSeparator, domain}));
+}
+
+void AIChatSettingsHelper::GetCustomModels(GetCustomModelsCallback callback) {
+  std::vector<mojom::CustomModelPtr> custom_models = GetCustomModels();
+  std::move(callback).Run(std::move(custom_models));
+}
+
+void AIChatSettingsHelper::AddCustomModel(mojom::CustomModelPtr model,
+                                          AddCustomModelCallback callback) {
+  if (!IsEndpointUrlValid(model->endpoint)) {
+    std::move(callback).Run(mojom::OperationResult::InvalidUrl);
+    return;
+  }
+
+  model_service_->AddCustomModel(std::move(model));
+  std::move(callback).Run(mojom::OperationResult::Success);
+}
+
+void AIChatSettingsHelper::SaveCustomModel(uint32_t index,
+                                           mojom::CustomModelPtr model,
+                                           SaveCustomModelCallback callback) {
+  if (!IsEndpointUrlValid(model->endpoint)) {
+    std::move(callback).Run(mojom::OperationResult::InvalidUrl);
+    return;
+  }
+
+  model_service_->SaveCustomModel(index, std::move(model));
+  std::move(callback).Run(mojom::OperationResult::Success);
+}
+
+void AIChatSettingsHelper::DeleteCustomModel(uint32_t index) {
+  model_service_->DeleteCustomModel(index);
+}
+
+void AIChatSettingsHelper::ChangeDefaultModel(const std::string& model_key) {
+  model_service_->ChangeDefaultModel(model_key);
+}
+
+void AIChatSettingsHelper::SetClientPage(
+    mojo::PendingRemote<mojom::SettingsPage> page) {
+  client_page_.Bind(std::move(page));
+}
+
+bool AIChatSettingsHelper::IsEndpointUrlValid(const GURL& url) {
```

Comment:
Please replace this with IsHTTPSOrLocalhostURL https://github.com/brave/brave-core/pull/24249/files#diff-3eb0d391f67f935b6c74e185966b350e70b62837c20a833de3a028b88c907c55R13

Reply:
done.